odoo/addons/base/models/ir_binary.py

```diff
@@ -45,7 +46,8 @@ def _find_record(
             record = self.env[res_model].browse(res_id).exists()
         if not record:
             raise MissingError(f"No record found for xmlid={xmlid}, res_model={res_model}, id={res_id}")
-
+        if access_token and verify_binary_token(record, field, access_token):
+            return record.sudo()
```

Comment:
Shouldn't token verification be done in `_find_record_check_access` ?

Reply:
The idea was to have the validation code next to the creation code in the same file, it's easier to maintain both together.

Similar to what is done for `hash_sign` and `verify_hash_signed`.

Reply:
Oh you mean the calling of the method.
`_find_record_check_access` has many overrides which call super in various order, usually afterwards.
When given, the token should come before all of them to avoid useless queries.